staging/src/k8s.io/legacy-cloud-providers/azure/azure_standard.go

```diff
@@ -273,12 +273,21 @@ func getBackendPoolName(ipv6DualStackEnabled bool, clusterName string, service *
 	return clusterName
 }
 
-func (az *Cloud) getLoadBalancerRuleName(service *v1.Service, protocol v1.Protocol, port int32, subnetName *string) string {
+func (az *Cloud) getLoadBalancerRuleName(service *v1.Service, protocol v1.Protocol, port int32) string {
 	prefix := az.getRulePrefix(service)
-	if subnetName == nil {
-		return fmt.Sprintf("%s-%s-%d", prefix, protocol, port)
+	ruleName := fmt.Sprintf("%s-%s-%d", prefix, protocol, port)
+	subnet := subnet(service)
+	if subnet == nil {
+		return ruleName
 	}
-	return fmt.Sprintf("%s-%s-%s-%d", prefix, *subnetName, protocol, port)
+
+	// Load balancer rule name must be less or equal to 80 charactors, so excluding the hyphen two segments cannot exceed 79
+	subnetSegment := *subnet
+	if len(ruleName) + len(subnetSegment) > 79 {
+		subnetSegment = subnetSegment[:79 - len(ruleName)]
+	}
+
+	return fmt.Sprintf("%s-%s-%s-%d", prefix, subnetSegment, protocol, port)
```

Comment:
is it possible that returned length of LoadBalancerRuleName exceeds 80?

Reply:
that's not possible, the length of prefix + protocol + port has been calculated at len(ruleName). And based on how we calculate, it won't exceed 80 at that time, because UID is 36 characters - 4 hythen + 1 'a' at begining, it's 33 characters. with ports and protocol, it's well below 80.

at L287 we use the rule name length to truncate it to 80.

Reply:
it won't since the subnetSegment has been truncated at L289